core/pkg/resourcehandler/localgitrepository.go

```diff
@@ -0,0 +1,107 @@
+package resourcehandler
+
+import (
+	"fmt"
+	"path"
+	"strings"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/config"
+	"github.com/go-git/go-git/v5/plumbing"
+)
+
+type LocalGitRepository struct {
+	repo   *git.Repository
+	head   *plumbing.Reference
+	config *config.Config
+}
+
+type GitCommit struct {
+	hash        string
+	authorName  string
+	authorEmail string
+	message     string
+	date        string
+}
+
+func NewLocalGitRepository(path string) (*LocalGitRepository, error) {
+	gitRepo, err := git.PlainOpen(path)
+	if err != nil {
+		return nil, err
+	}
+
+	head, err := gitRepo.Head()
+	if err != nil {
+		return nil, err
+	}
+
+	if !head.Name().IsBranch() {
+		return nil, fmt.Errorf("current HEAD reference is not a branch")
+	}
+
+	config, err := gitRepo.Config()
+	if err != nil {
+		return nil, err
+	}
+
+	return &LocalGitRepository{
+		repo:   gitRepo,
+		head:   head,
+		config: config,
+	}, nil
+}
+
+func (g *LocalGitRepository) GetBranchName() string {
+	return g.head.Name().Short()
+}
+
+func (g *LocalGitRepository) GetOriginUrl() string {
+	branchName := g.GetBranchName()
+	if branchRef, branchFound := g.config.Branches[branchName]; branchFound {
+		remoteName := branchRef.Remote
+		return g.config.Remotes[remoteName].URLs[0]
```

Comment:
Are you sure we don't need to check the list length before accessing it?

Reply:
So I followed the API documentation - they note that "it must be non-empty": https://github.com/go-git/go-git/blob/f0b111ab70e4e90013658b0835929b2083902017/config/config.go#L550

And I have seen that they also access the first URL without checking in other functions: https://github.com/go-git/go-git/blob/f0b111ab70e4e90013658b0835929b2083902017/config/config.go#L642

If you think we should check this on our side I will add a check